### ISSUE TITLE
Support for `eth_createAccessList`

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/web3.py#2258]
### What was wrong?

Unsure if this is standardized but, support for `eth_createAccessList` would be useful

https://geth.ethereum.org/docs/rpc/ns-eth#eth_createaccesslist

### How can it be fixed?

Add support for that JSON-RPC endpoint either in the `w3.geth` namespace or under `w3.eth` if it is standardized (I think it is)

**Comment by Harmouch101** [comment]
Hey there, I think I can work on dis issue. However, after digging a bit deeper, I realized there is a bit of struggle in implementing the formatter for the response. From what I understood from the [eip-2930 example](https://eips.ethereum.org/EIPS/eip-2930), I think it should be something along these lines:

```python
ACCESS_LIST_FORMATTER = apply_list_to_array_formatter({
    'address': to_checksum_address,
    'storageKeys': apply_list_to_array_formatter(to_hexbytes(32))
})

ACCESS_LIST_RESPONCE_FORMATTER = apply_formatters_to_dict({
    "accessList": ACCESS_LIST_FORMATTER,
    "gasUsed": to_integer_if_hex,
})
```

And under the Eth namespace formatters, we need to add the following line:

```python
    RPC.eth_createAccessList: ACCESS_LIST_RESPONCE_FORMATTER,
```

Similarly, for the request, the formatter might look like:

```python
    RPC.eth_createAccessList: apply_formatter_at_index(transaction_param_formatter, 0),
```

Lemme know whether or not I misinterpreted the format of the req/res to continue working on dis issue.

Regards.

**Comment by kclowes** [comment]
closed via #2381
